spec/features/school_member/listing_school_members_spec.rb

```diff
@@ -0,0 +1,137 @@
+
+  it 'responds with the class members JSON array' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+
+    expect(data.size).to eq(4)
+  end
+
+  it 'responds with the correct student ids' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+    student_ids = data.pluck(:student).compact.pluck(:id)
+
+    students.each do |student|
+      expect(student_ids).to include(student.id)
+    end
+  end
+
+  it 'responds with the expected student parameters' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+    student_data = data.pluck(:student).compact.find { |member| member[:id] == students[0].id }
+
+    expect(student_data).to eq(
+      {
+        id: students[0].id,
+        username: students[0].username,
+        name: students[0].name
+      }
+    )
+  end
+
+  it 'responds with the correct teacher id' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+    teacher_id = data.pluck(:teacher).compact.pick(:id)
+
+    expect(teacher_id).to eq(teacher.id)
+  end
+
+  it 'responds with the expected teacher parameters' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+    teacher_data = data.pluck(:teacher).compact
+
+    expect(teacher_data.first).to eq(
+      {
+        id: teacher.id,
+        name: teacher.name,
+        email: teacher.email
+      }
+    )
+  end
+
+  it 'responds with teachers at the top' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+
+    expect(data[0][:teacher]).to be_truthy
+  end
+
+  it 'responds with students in alphabetical order by name ascending' do
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+
+    student_names = data.pluck(:student).compact.pluck(:name)
+    sorted_student_names = student_names.sort
+
+    expect(student_names).to eq(sorted_student_names)
+  end
+
+  it "responds with nil attributes for students if the user profile doesn't exist" do
+    stub_user_info_api_for_unknown_users(user_id: students.first.id)
+
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+
+    expect(data.first[:student_name]).to be_nil
+  end
+
+  it 'does not include class members that belong to a different class' do
+    student = create(:student, school:)
+    different_class = create(:school_class, school:, teacher_id: teacher.id)
+    create(:class_member, school_class: different_class, student_id: student.id)
+
+    get("/api/schools/#{school.id}/members", headers:)
+    data = JSON.parse(response.body, symbolize_names: true)
+
+    expect(data.size).to eq(4)
+  end
```

Comment:
❓ a bit confused by this because it's a different school right but this seems to make a new student in the same school and they're not included in the response? 🤔 teachers/owners listing school members should see everyone

Reply:
you're quite right, I was a bit trigger happy on the old copy and paste